Code/GraphMol/Substruct/SubstructUtils.h

```diff
@@ -27,6 +27,16 @@ RDKIT_SUBSTRUCTMATCH_EXPORT bool chiralAtomCompat(const Atom* a1,
                                                   const Atom* a2);
 RDKIT_SUBSTRUCTMATCH_EXPORT bool bondCompat(const Bond* b1, const Bond* b2,
                                             const SubstructMatchParameters& ps);
+//! This postprocesses the passed substruct matches and returns
+//! the match that has the largest number of non-hydrogen atoms
+//! in correspondence of terminal dummy atoms
+RDKIT_SUBSTRUCTMATCH_EXPORT std::vector<MatchVectType>::const_iterator
```

Comment:
returning an iterator here feels somehow wrong: it complicates client code and raises the question: "what does it mean if I increment/decrement the result of this function?"

How about simplifying things by returning a `const MatchVectType &`?

Reply:
Yeah, of course, that's a leftover of when the function was not publicly accessible, that needs to be changed now. Done.